core/spec.md

```diff
@@ -146,7 +146,8 @@ TODO: Add `format`
 - Type: String   # SHOULD this be a URI-Reference?
 - Description: An immutable unique identifier of the entity.
 - Constraints:
-  - MUST be a non-empty string
+  - MUST be a non-empty string that is compliant with the definition of
+    an HTTP header [`field-name`][field-name]`.
```

Comment:
I believe this prohibits the use of `/` and `:` within an ID (as well as various other characters, of course). I can see those being pretty common within IDs... do we definitely want to prohibit them?

Reply:
oh wait - you're right, I messed up. I need to worry about valid http header values, not names. Doi! Will fix